src/poprox_recommender/components/embedders/caption_embedder.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from dataclasses import dataclass
+from os import PathLike
+from typing import List
+
+import torch as th
+from safetensors.torch import load_file
+from transformers import AutoTokenizer
+
+from poprox_recommender.model import ModelConfig
+from poprox_recommender.model.nrms.news_encoder import NewsEncoder
+from poprox_recommender.paths import model_file_path
+from poprox_recommender.pytorch.decorators import torch_inference
+
+logger = logging.getLogger(__name__)
+CAPTION_LENGTH_LIMIT = 30
+
+
+@dataclass
+class CaptionEmbedderConfig:
+    model_path: PathLike
+    device: str | None
+
+
+class CaptionEmbedder:
+    config: CaptionEmbedderConfig
+    model: NewsEncoder
+    tokenizer: AutoTokenizer
+    embedding_cache: dict[str, th.Tensor]
+
+    def __init__(self, config: CaptionEmbedderConfig | None = None, **kwargs):
+        if config is None:
+            config = CaptionEmbedderConfig(
+                model_path=model_file_path("nrms-mind/news_encoder.safetensors"), device="cpu"
+            )
+
+        self.config = config
+
+        checkpoint = load_file(self.config.model_path)
```

Comment:
Loading the model here manually would work, but there's a better way: you can use the `NRMSArticleEmbedder` component directly, which knows how to load the model and will be cached to avoid loading it more than once.

Here's an example of what that looks like: https://github.com/CCRI-POPROX/poprox-recommender/blob/main/src/poprox_recommender/components/embedders/topic_wise_user.py#L156-L158